### ISSUE TITLE
Return `Result` from syscalls that can be unsuccessful

### ISSUE THREAD

**Issue by shamilsan** [gear-tech/gear#314]
### Problem to Solve

Now we can't get the `msg::send` or `msg::reply` result while writing programs using `gstd`. Furthermore, we can fall down in the infinite waiting after say `send_and_wait` to invalid address.

### Possible Solution

Functions that may be unsuccesful should return `Result`.

```rust
enum MessageError {
    InvalidActorId,
    OutOfGas,
    ...
}

type MessageResult<T> = Result<T, MessageError>;

msg::send(...) -> MessageResult<MessageId>;
msg::send_commit(...) -> MessageResult<MessageId>;
msg::reply(...) -> MessageResult<MessageId>;
msg::reply_commit(...) -> MessageResult<MessageId>;
msg::send_and_wait(...).await -> MessageResult<Vec<u8>>;
```

### Notes

Consider other syscalls for returning `Result` too.

**Comment by NikVolf** [comment]
`send_and_wait` can't decide if the address is valid or not
there is no such thing as invalid actor id

**Comment by shamilsan** [comment]
> `send_and_wait` can't decide if the address is valid or not there is no such thing as invalid actor id

What we get if we `send_and_wait` to the address that doesn't exists?

**Comment by NikVolf** [comment]
If no program exists at the specific address, it is treated as user address 

And any user address exists as it is a valid public key

**Comment by NikVolf** [comment]
But indeed `send_and_wait` can ultimately resolve as a future error

When message spends more than one block in waiting list of something that can't afford the storage (be that program or user)

**Comment by breathx** [comment]
Seems like fixed via #401, isn't it?
In case of gas limit exceeded for message in wait-list, it should produce reply trap message for the sender contract - this scenario will be handled well.

**Comment by shamilsan** [comment]
Agree, we can expand the set of exit codes later.
